tools/PI/DevHome.PI/BarWindowHorizontal.xaml.cs

```diff
@@ -227,6 +227,16 @@ private void ManageExternalToolsButton_Click(object sender, RoutedEventArgs e)
         ExpandedViewControl.NavigateToSettings(typeof(AdditionalToolsViewModel).FullName!);
     }
 
+    private void ExternalToolsMenu_Opening(object sender, object e)
+    {
+        // Cancel the opening of the menu if there are no items.
+        var flyout = sender as MenuFlyout;
+        if (flyout is not null && flyout.Items is not null && flyout.Items.Count == 0)
```

Comment:
This can be simplified to `if (sender is MenuFlyout flyout && flyout?.Items?.Count == 0)`